Drop-In/src/main/java/com/braintreepayments/api/dropin/adapters/VaultedPaymentMethodsAdapter.java

```diff
@@ -21,11 +22,14 @@
 
     private PaymentMethodNonceCreatedListener mSelectedListener;
     private List<PaymentMethodNonce> mPaymentMethodNonces;
+    private BraintreeFragment mBraintreeFragment;
```

Comment:
Ideally views/adapters shouldn't know about any other objects other than the ones they are presenting.

I don't have a super straight forward suggestion for you guys. You could add a listener that sends analytic events and chains the PaymentMethodNonceCreatedListeners. Here is some psuedo code for this: 

```
// Add field in DropInActivity.java
private final PaymentMethodNonceCreatedListener vaultedPaymentMethodSelected = (paymentMethodNonce) -> {
  mBraintreeFragment.sendAnalyticsEvent("vaulted-card.selected")

  DropInActivity.this.onPaymentMethodNonceCreated(paymentMethodNonce)
};

...

mVaultedPaymentMethodsView.setAdapter(vaultedPaymentMethodSelected, ...)
```

Reply:
We found another workaround. We instead including this analytic event on the already existing `onPaymentMethodNonceCreated` function. Check it out [here](https://github.com/braintree/braintree-android-drop-in/pull/130/commits/54272e039a48f921b7bbc89a1a3bd93b4cff20f8#diff-4253f3b4ed47a06cee52c9f47fc90b04R203). 

It fires as expected.

Reply:
Resolved in [this commit](https://github.com/braintree/braintree-android-drop-in/pull/130/commits/d5d48da09ca22f9b928193056ebd66b38328b50a).